akka-actor-typed/src/main/scala/akka/actor/typed/internal/TimerSchedulerImpl.scala

```diff
@@ -21,8 +21,16 @@ import scala.reflect.ClassTag
  * INTERNAL API
  */
 @InternalApi private[akka] object TimerSchedulerImpl {
+  trait TimerMsg {
+    def key: Any
+    def generation: Int
+    def owner: AnyRef
+  }
```

Comment:
Should it be sealed just to show it isn't meant impls outside this object?

Reply:
Agree that sealed would help here, can apply this and merge